app/modules/google/google.sagas.js

```diff
@@ -0,0 +1,35 @@
+import { call, put, takeLatest } from 'redux-saga/effects';
+import envConfig from 'env-config';
+
+import { GoogleActions, GoogleTypes } from './google.redux';
+import { get } from '../api/api.sagas';
+
+
+export function* getPlacesSaga({ latitude, longitude, placeType, name }) {
+  try {
+    const data = yield call(get, 'api/places', {
+      key: envConfig.google.apiKey,
+      location: `${latitude},${longitude}`,
+      radius: 10000,
+      type: placeType,
+      name,
+    });
+
+    yield put(GoogleActions.getPlacesSuccess(data));
+  } catch (e) {
+    yield put(GoogleActions.getPlacesFailure(e));
+  }
+}
+
+export function* choosePlacesSaga() {
+  if (Math.random() > 0.5) {
```

Comment:
a mi się wydaje, że to jest idealny przykład przewagi sagi nad thunkiem, w którym dla tego scenariusza mielibyśmy już ładny łańcuszek callbacków 😄